core/flyout_base.ts

```diff
@@ -626,8 +627,11 @@ export abstract class Flyout extends DeleteArea implements IFlyout {
     const parsedContent = toolbox.convertFlyoutDefToJsonArray(flyoutDef);
     const flyoutInfo = this.createFlyoutInfo(parsedContent);
 
+    renderManagement.triggerQueuedRenders();
```

Comment:
> Would it make sense to make show async in the future

Seems reasonable.

> If so I can set this up to return a resolved promise to make it easier for folks to transition.

by "this" do you mean the `show` method? similar to my comments below, I feel like that's a bit misleading right now. also it's kind of hard to communicate that they can go ahead and wait on a promise now. if at some point we do a breaking change where we make show async, then that's when it would make the most sense to announce the change, so idk what we'd get out of adding an immeditately-resolved promise right now